spki/src/spki.rs

```diff
@@ -30,16 +38,20 @@ use crate::AlgorithmIdentifier;
 /// ```
 ///
 /// [RFC 5280 § 4.1.2.7]: https://tools.ietf.org/html/rfc5280#section-4.1.2.7
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub struct SubjectPublicKeyInfo<'a> {
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SubjectPublicKeyInfo<Params, Key> {
     /// X.509 [`AlgorithmIdentifier`] for the public key type
-    pub algorithm: AlgorithmIdentifierRef<'a>,
+    pub algorithm: AlgorithmIdentifier<Params>,
 
     /// Public key data
-    pub subject_public_key: &'a [u8],
+    pub subject_public_key: Key,
 }
 
-impl<'a> SubjectPublicKeyInfo<'a> {
+impl<'a, Params, Key> SubjectPublicKeyInfo<Params, Key>
+where
+    Params: Choice<'a> + Encode,
+    Key: BitStringLike<'a>,
```

Comment:
The `BitStringLike` trait seems like an unnecessary addition to me.

Instead you can do:

```suggestion
    Key: Decode<'a> + Encode + FixedTag<TAG = Tag::BitString>,
```

...which accomplishes the same thing without introducing an additional trait

Reply:
it doesn't appear to be possible because `Tag::BitString` isn't a type itself

Reply:
Aah, here's the relevant issue: https://github.com/rust-lang/rust/issues/92827

Perhaps you can just use a bound like this:

```toml
Key: Decode<'a> + Encode + FixedTag
```

...and leave a note to add the `FixedTag<TAG = { Tag::BitString }` when the above issue is resolved?

Reply:
works for me.